src/engine/IOrdered.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include <string>
+
+namespace admirals {
+class IOrdered {
+public:
+    virtual std::string name() const = 0;
+    virtual float order() const = 0;
+};
+} // namespace admirals
```

Comment:
I think name and order should be member variables of IOrdered. Feels weird having the virtual getters that (presumably) gets those values from the sub-class.

Reply:
A solution would be to have a constructor for IOrdered which must take in a value for name and order, that way all information is stored in the class without having to get it from the sub-class.

Reply:
Lets discuss this in the meeting today.